### ISSUE TITLE
Can't upgrade tfbridge to v3.33.0 or higher due to azcore conflict

### ISSUE THREAD

**Issue by viveklak** [pulumi/pulumi-vault#197]
### What happened?

This is a follow-up from #192 for which a workaround was added in #196. However, we are currently blocked from taking a new version of the bridge because updating pulumi-terraform-bridge to v3.33.0 causes a build failure:

```
github.com/hashicorp/terraform-provider-vault/internal/provider
../../../go/pkg/mod/github.com/pulumi/terraform-provider-vault@v1.9.1-0.20221013032412-e762e637392b/internal/provider/auth_azure.go:185:9: tOpts.TenantID undefined (type policy.TokenRequestOptions has no field or method TenantID)
# github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:19:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:62:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:105:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:148:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:191:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:234:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:277:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:320:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:363:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:406:13: undefined: azcore.Poller
../../../go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources@v0.3.1/zz_generated_pollers.go:406:13: too many errors
```

The underlying issue seems to be because of these updates (going from tfbridge 3.32.0 to 3.33.0):
```
go: downloading github.com/pulumi/pulumi-terraform-bridge/v3 v3.33.0
go: upgraded cloud.google.com/go v0.102.1 => v0.103.0
go: upgraded github.com/Azure/azure-sdk-for-go v61.4.0+incompatible => v66.0.0+incompatible
go: upgraded github.com/Azure/azure-sdk-for-go/sdk/azcore v0.22.0 => v1.1.1
go: upgraded github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.13.2 => v1.0.0
go: upgraded github.com/Azure/azure-sdk-for-go/sdk/internal v0.9.1 => v1.0.0
go: upgraded github.com/Azure/go-autorest/autorest v0.11.24 => v0.11.28
go: upgraded github.com/Azure/go-autorest/autorest/adal v0.9.18 => v0.9.21
go: upgraded github.com/Azure/go-autorest/autorest/azure/cli v0.4.5 => v0.4.6
go: upgraded github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a => v0.0.0-20210307081110-f21760c49a8d
```

This is because of the bump of pulumi/pkg etc. to `v3.44.2` which brought in new versions of the above package in https://github.com/pulumi/pulumi/pull/10900/.

Vault terraform provider is still using the older versions of the above libraries which causes the conflict.

### Steps to reproduce

```
cd provider ;  go get -u github.com/pulumi/pulumi-terraform-bridge/v3; make build
```

### Expected Behavior

Build succeeds

### Actual Behavior

Build fails

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by jazzyfresh** [comment]
encountered this in pulumi-consul as well, but for the `tfexec` package

**Comment by t0yv0** [comment]
I've bumped into this today again and tried a few resolutions.

Unfortunately upstream provider is very much behind the latest Azure SDKs.

```
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.22.0
-	github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.13.2
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources v0.3.1
-	github.com/Azure/go-autorest/autorest v0.11.24
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.7.2
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.1
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources v1.1.1
```

I've tried upgrading the provider on our end, that is possible, azureAccessCredentialsDataSourceRead function needs to be rewritten against new Azure SDK APIs, that looked like a day of work with testing, not entirely trivial. 

The other way to solve this is if we could get a form of https://github.com/pulumi/pulumi/issues/12139 

The source of the conflict now is a diamond dependency where we are inheriting one version of azcore through upstream, and another version of azcore through a transitive dependency to pulumi filestate,  which is coupled to pulumi/pkg dependency, but is not really in use for this project. Breaking that cycle somehow so that bridge and bridged providers do not pull in ascore dependency would pave a way forward and be generally beneficial.
